tools/python/util/convert_onnx_models_to_ort.py

```diff
@@ -74,15 +75,15 @@ def _create_session_options(
 
 def _convert(
     model_path_or_dir: pathlib.Path,
-    output_dir: typing.Optional[pathlib.Path],
+    output_dir: pathlib.Path | None,
     optimization_level_str: str,
     optimization_style: OptimizationStyle,
     custom_op_library: pathlib.Path,
     create_optimized_onnx_model: bool,
     allow_conversion_failures: bool,
     target_platform: str,
-    session_options_config_entries: typing.Dict[str, str],
-) -> typing.List[pathlib.Path]:
+    session_options_config_entries: dict[str, str],
```

Comment:
do the type annotations with `dict` and `list` work with older Python versions (whatever the minimum we support is)?

Reply:
by default, no.

But with `from __future__ import annotations`, it'll work with all versions of python we support.